cdap-app-templates/cdap-etl/cdap-etl-tools/src/main/java/co/cask/cdap/etl/tool/UpgradeTool.java

```diff
@@ -0,0 +1,284 @@
+    if (!configFile.exists()) {
+      throw new IllegalArgumentException(configFilePath + " does not exist.");
+    }
+    if (!configFile.isFile()) {
+      throw new IllegalArgumentException(configFilePath + " is not a file.");
+    }
+    String fileContents = new String(Files.readAllBytes(configFile.toPath()), StandardCharsets.UTF_8);
+
+    ETLAppRequest artifactFile = GSON.fromJson(fileContents, ETLAppRequest.class);
+    if (!shouldUpgrade(artifactFile.artifact)) {
+      throw new IllegalArgumentException(
+        "Cannot update for artifact " + artifactFile.artifact + ". " +
+          "Please check the artifact is cdap-etl-batch or cdap-etl-realtime in the system scope of version 3.2.x.");
+    }
+
+    String version = getVersion();
+
+    if (BATCH_NAME.equals(artifactFile.artifact.getName())) {
+      ArtifactSummary artifact = new ArtifactSummary(BATCH_NAME, version, ArtifactScope.SYSTEM);
+      OldETLBatchConfig oldConfig = GSON.fromJson(artifactFile.config, OldETLBatchConfig.class);
+      AppRequest<ETLBatchConfig> updated = new AppRequest<>(artifact, oldConfig.getNewConfig());
+      System.out.println(GSON.toJson(updated));
+    } else {
+      ArtifactSummary artifact = new ArtifactSummary(REALTIME_NAME, version, ArtifactScope.SYSTEM);
+      OldETLRealtimeConfig oldConfig = GSON.fromJson(artifactFile.config, OldETLRealtimeConfig.class);
+      AppRequest<ETLRealtimeConfig> updated = new AppRequest<>(artifact, oldConfig.getNewConfig());
+      System.out.println(GSON.toJson(updated));
+    }
+  }
+
+  // reads current version from the etl.properties file contained in resources.
+  private static String getVersion() {
+    Properties prop = new Properties();
+    try {
+      URL resource = UpgradeTool.class.getResource("/etl.properties");
+      try (InputStream in = UpgradeTool.class.getResourceAsStream("/etl.properties")) {
+        prop.load(in);
+      }
+      return prop.getProperty("version");
+    } catch (Exception e) {
+      throw new RuntimeException("Error determining version. Please check that the jar was built correctly.", e);
+    }
+  }
+
+  // class used to deserialize the old pipeline requests
+  @SuppressWarnings("unused")
+  private static class ETLAppRequest {
+    private ArtifactSummary artifact;
+    private JsonObject config;
+  }
+
+  private static boolean shouldUpgrade(ArtifactSummary artifactSummary) {
+    // check its the system etl artifacts
+    if (artifactSummary.getScope() != ArtifactScope.SYSTEM) {
+      return false;
+    }
+
+    if (!BATCH_NAME.equals(artifactSummary.getName()) && !REALTIME_NAME.equals(artifactSummary.getName())) {
+      return false;
+    }
+
+    // check its 3.2.x versions of etl
+    ArtifactVersion artifactVersion = new ArtifactVersion(artifactSummary.getVersion());
+    return !(artifactVersion.getMajor() == null || artifactVersion.getMinor() == null) &&
+      artifactVersion.getMajor() == 3 && artifactVersion.getMinor() == 2;
```

Comment:
might be useful to print if the version is 3.3, we can say that artifact {artifact-id} has been already upgraded.